src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -251,8 +254,31 @@ protected virtual async Task<RecordValue> FindAsync(RecordValue baseRecord, Canc
         }
 
         protected static async Task<bool> MatchesAsync(RecordValue currentRecord, RecordValue baseRecord, CancellationToken cancellationToken)
-        {
-            var ret = true;
+        {            
+            if (currentRecord.TryGetPrimaryKey(out string currentRecordPrimaryKeyValue))
+            {
+                string pKey = currentRecord.GetPrimaryKeyName();
+
+                if (string.IsNullOrEmpty(pKey))
+                {

```

Comment:
just curious: what's an example that hits the true/false branches?

Reply:
if you have your own RecordValue and don't override GetPrimaryKeyName